### PR TITLE
Fixing issue when the parent is a UserControl #15

### PR DESCRIPTION
**Pull request by HerrKater**
The WizardControl assumes that the parent is always a Form. See WizardControl.OnParentChanged line 548. This prevents the ConfigureWindowFrame() from running in case the parent is a UserControl. Since the Form and the UserControl doesn't have a common base class that defines the Load event - which is triggering the previously mentioned function - we have to hook to both of those separate Load events, which makes the solution less elegant, but also fixes a nullreference exception.

### CODE REVIEW

**Comment by dahall** [comment]
Has this been tested as both a child of a `Form` and a child of `UserControl`?

File: src/Library/WizardControl.cs [hunk 1]
```diff
@@ -43,7 +43,10 @@ public partial class WizardControl :
 		private WizardClassicStyle classicStyle = WizardClassicStyle.AeroStyle;
 		private Point formMoveLastMousePos;
 		private bool formMoveTracking;
-		private Form parentForm;
+		private ContainerControl parentControl;
+		private Form parentAsForm => parentControl as Form;
```
Comment (dahall) [inline src/Library/WizardControl.cs:47]: Please remove these two private properties that really just hide the function of checking the control type. See other comments for handling.

File: src/Library/WizardControl.cs [hunk 2]
```diff
@@ -400,12 +403,17 @@ private void RemoveSystemEvents()
 		protected override void OnParentChanged(EventArgs e)
 		{
 			base.OnParentChanged(e);
-			if (parentForm != null)
-				parentForm.Load -= parentForm_Load;
-			parentForm = Parent as Form; // FindForm();
+			if (parentAsForm!=null)
```
Comment (dahall) [inline src/Library/WizardControl.cs:406]: `if (parentControl is Form parentAsForm)`

File: src/Library/WizardControl.cs [hunk 3]
```diff
@@ -495,19 +503,22 @@ private void ConfigureWindowFrame()
 				titleBar.BackColor = commandArea.BackColor;
 			}
 
-			if (parentForm == null) return;
+			if (parentControl == null) return;
 			if (!SuppressParentFormCaptionSync)
-				parentForm.Text = Title;
-			if (!SuppressParentFormIconSync && titleImageIcon != null)
+				parentControl.Text = Title;
+			if (parentAsForm != null)
```
Comment (dahall) [inline src/Library/WizardControl.cs:509]: `if (parentControl is Form parentAsForm)`

File: src/Library/WizardControl.cs (hunk 1 above)
Comment (HerrKater) [reply src/Library/WizardControl.cs:47]: Fixed, I accidentally created a new pull request. This one can be rejected.

File: src/Library/WizardControl.cs (hunk 2 above)
Comment (HerrKater) [reply src/Library/WizardControl.cs:406]: Fixed, I accidentally created a new pull request. This one can be rejected.

File: src/Library/WizardControl.cs (hunk 3 above)
Comment (HerrKater) [reply src/Library/WizardControl.cs:509]: Fixed, I accidentally created a new pull request. This one can be rejected.

**Comment by dahall** [comment]
Closing as duplicate with #90

**Pull request closed without merging**